SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -301,11 +336,19 @@ - (void)didEnterStateSettingUpHMI {
         return;
     }
 
+    if ([self sdl_didAccessoryDisconnect]) {
+        return;
+    }
+
     // We are sure to have a HMIStatus, set state to ready
     [self.lifecycleStateMachine transitionToState:SDLLifecycleStateReady];
 }
 
 - (void)didEnterStateReady {
+    if ([self sdl_didAccessoryDisconnect]) {
```

Comment:
Are we certain having these everywhere is necessary? We are getting a transport disconnect notification (transportDidDisconnect) if the accessory disconnected.